sdk/python/lib/pulumi/automation/_remote_workspace.py

```diff
@@ -185,12 +185,14 @@ def _create_local_workspace(
     opts: Optional[RemoteWorkspaceOptions] = None,
 ) -> LocalWorkspace:
 
-    if commit_hash is not None and branch is not None:
-        raise Exception("commit_hash and branch cannot both be specified.")
-    if commit_hash is None and branch is None:
-        raise Exception("at least commit_hash or branch are required.")
+    if not url:
+        raise Exception("url is required.")
+    if branch and commit_hash:
+        raise Exception("branch and commit_hash cannot both be specified.")
+    if not branch and not commit_hash:
+        raise Exception("either branch or commit_hash is required.")
```

Comment:
Minor change to reorder the check to match the error message (changed to be consistent with CLI and other languages).